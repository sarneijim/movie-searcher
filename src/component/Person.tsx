import React, { Component } from "react";
import Typography from "@material-ui/core/Typography";
import styles from "../styles/details.module.css";
import { imgPath, personRequest } from "../request";

export default class Actor extends Component<{}, any> {
  constructor(props: any) {
    super(props);
    this.state = { data: [] };
    if (props.match?.params?.id) {
      this.httpRequest(props.match.params.id);
    }
  }

  httpRequest = async (id: string) => {
    try {
      var response = await personRequest(id);
      this.setState({ data: response });
    } catch (err) {
      console.error(err);
    }
  };

  render() {
    return (
      <div className={styles.root}>
        <div className={styles.movie}>
          <div className={styles.left}>
            <img
              className={styles.img}
              src={`${imgPath}${this.state.data.profile_path}`}
              alt={this.state.data.title}
            />
          </div>
          <div className={styles.right}>
            <Typography component="h4" variant="h4">
              {this.state.data.name}
            </Typography>
            <ul>
              <div>
                <pre>{JSON.stringify(this.state.data, null, 2)}</pre>
              </div>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}
