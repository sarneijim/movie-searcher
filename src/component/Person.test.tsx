import renderer from "react-test-renderer";
import Person from "./Person";

describe("Person", () => {
  let component: any;
  beforeEach(() => {
    component = renderer.create(<Person></Person>);
  });

  afterEach(() => {
    component = null;
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match the snapshot", () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("check httpRequest method", () => {
    let instance: any = component.getInstance();
    const utils = require("../request");
    const personRequestSpy = jest
      .spyOn(utils, "personRequest")
      .mockReturnValue({ id: 123 });
    instance.httpRequest("abc");
    expect(personRequestSpy).toHaveBeenCalledWith("abc");
    personRequestSpy.mockClear();
  });
});
