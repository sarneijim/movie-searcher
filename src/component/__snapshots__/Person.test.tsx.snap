// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot 1`] = `
<div
  className="root"
>
  <div
    className="movie"
  >
    <div
      className="left"
    >
      <img
        className="img"
        src="https://image.tmdb.org/t/p/w500undefined"
      />
    </div>
    <div
      className="right"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      />
      <ul>
        <div>
          <pre>
            []
          </pre>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
